packages/interactivity-router/src/full-csn.js

```diff
@@ -0,0 +1,200 @@
+
+	// Detect nodes that should be added or removed.
+	for ( const child of document.head.children ) {
+		const id = getTagId( child );
+		// Always remove styles and links as they might change.
+		if ( child.nodeName === 'LINK' || child.nodeName === 'STYLE' )
+			toRemove.push( child );
+		else if ( newHeadMap.has( id ) ) newHeadMap.delete( id );
+		else if ( child.nodeName !== 'SCRIPT' && child.nodeName !== 'META' )
+			toRemove.push( child );
+	}
+
+	// Prepare new assets.
+	const toAppend = [ ...newHeadMap.values() ];
+
+	// Apply the changes.
+	toRemove.forEach( ( n ) => n.remove() );
+	document.head.append( ...toAppend );
+};
+
+const nextTick = ( fn ) =>
+	new Promise( ( resolve ) => setTimeout( () => resolve( fn() ) ) );
+
+// Fetch head assets of a new page.
+const fetchAssets = async ( document ) => {
+	const headTags = [];
+	const assets = [
+		{
+			tagName: 'style',
+			selector: 'link[rel=stylesheet]',
+			attribute: 'href',
+		},
+		{ tagName: 'script', selector: 'script[src]', attribute: 'src' },
+	];
+	for ( const asset of assets ) {
+		const { tagName, selector, attribute } = asset;
+		const tags = document.querySelectorAll( selector );
+
+		// Use Promise.all to wait for fetch to complete
+		await Promise.all(
+			Array.from( tags ).map( async ( tag ) => {
+				const attributeValue = tag.getAttribute( attribute );
+				if ( ! headElements.has( attributeValue ) ) {
+					const response = await fetch( attributeValue );
+					const text = await response.text();
+					headElements.set( attributeValue, {
+						tag,
+						text,
+					} );
+				}
+
+				const headElement = headElements.get( attributeValue );
+				const element = document.createElement( tagName );
+				element.innerText = headElement.text;
+				for ( const attr of headElement.tag.attributes ) {
+					element.setAttribute( attr.name, attr.value );
+				}
+				headTags.push( element );
+			} )
+		);
+	}
+
+	return [
+		document.querySelector( 'title' ),
+		...document.querySelectorAll( 'style' ),
+		...headTags,
+	];
+};
+
+// Fetch a new page and convert it to a static virtual DOM.
+const fetchPage = async ( url ) => {
+	const html = await window.fetch( url ).then( ( r ) => r.text() );
+	if ( ! canDoClientSideNavigation() ) return false;
+	const dom = new window.DOMParser().parseFromString( html, 'text/html' );
+	const head = await fetchAssets( dom );
+	return { head, body: toVdom( dom.body ) };
+};
+
+// Check if the link is valid for client-side navigation.
+const isValidLink = ( ref ) =>
+	ref &&
+	ref instanceof window.HTMLAnchorElement &&
+	ref.href &&
+	( ! ref.target || ref.target === '_self' ) &&
+	ref.origin === window.location.origin;
+
+// Check if the event is valid for client-side navigation.
+const isValidEvent = ( event ) =>
+	event.button === 0 && // Left clicks only.
+	! event.metaKey && // Open in new tab (Mac).
+	! event.ctrlKey && // Open in new tab (Windows).
+	! event.altKey && // Download.
+	! event.shiftKey &&
+	! event.defaultPrevented;
```

Comment:
Is this variable covering a user can open a link in a new tab by pressing the mouse wheel?

Reply:
I've tested it, and it seems to be working. From my tests, it seems that when you press the mouse wheel, it doesn't trigger an onClick event, so `actions.navigate` is not called, and it just works as expected.